src/v/cloud_topics/core/pipeline_stage.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "utils/named_type.h"
+
+namespace experimental::cloud_topics::core {
+
+/// Processing stage id
+using pipeline_stage = named_type<int, struct _pipeline_stage_tag>;
```

Comment:
IMHO this should be a class with a private constructor with pipeline declared as friend.

Much clearer indicates that it is a opaque type and no one should mess with it. named_type allows reading it, modifying it, constructing arbitrary values.

Also, should be squshed with the commit which uses this type.

Reply:
> Much clearer indicates that it is a opaque type and no one should mess with it. named_type allows reading it, modifying it, constructing arbitrary values.

Reply:
the question was about squashing

Reply:
I understand the motivation behind opaque type. It's just not obvious to me that we should do this. THe issue is quite minor. The code is used only internally. There will be like 3 or 4 pipeline stages inside the cloud_topics package and no external users of this stuff.

Reply:
I also extremely dislike friendship over long distances. If something lives in a different module it shouldn't be a friend. I'll think about some other approach though.